ConcurrentAPI.adoc

```diff
@@ -0,0 +1,244 @@
+== Distributed Concurrent API
+
+=== Motivation
+
+Coming from an initial issue where we need to provide an implementation for vert.x cluster manager for the shared data feature,
+provide concurrent and distributed programming capabilities.
+
+* Distributed Locks
+* Distributed CountDownLatch
+
+=== Scope
+
+All these interfaces should be placed in commons, in order to be used either in Embedded and Server mode.
+
+== Infinispan Distributed Locks
+
+=== DistributedLockManager and configuration
+
+```java
+package org.infinispan.lock.api;
+
+public class DistributedLockManager {
+   boolean defineLock(LockConfiguration configuration);
+   InfinispanLock get(String name);
+   LockConfiguration getConfiguration(String name);
+   boolean isDefined(String name);
+   CompletableFuture<Boolean> remove(String name);
+}
+
+
+public class LockConfiguration {
+   private final String name; // default "ISPN-Lock"
+   private final boolean rentrant; // default true
+   private final boolean silentFailover; // default true
+}
+
+```
+###### InfinispanLock defineLock(LockConfiguration configuration)
+
+Defines a lock with the specific name and LockConfiguration.
+It does not overwrite existing configurations.
+
+Returns true if successfully defined or false if the counter exists or any other failure.
+If silentFailover is false, then InfinispanLockException will be raised.
+
+###### InfinispanLock get(String name)
+
+Get's a InfinipanLock by it's name and creates it if does not exist with the default configuration.
```

Comment:
Hmmmm, I guess a lock is a not a cache with it's start/stop lifecycle methods... I'm happy to keep names consistent with CounterManager.